### PR TITLE
util-lib: wrap personality() to fix up broken glibc error handling

### PR DESCRIPTION
**Pull request by poettering**
glibc appears to propagate different errors in different ways, let's fix
this up, so that our own code doesn't get confused by this.

See #6752 + #6737 for details.

Fixes: #6755

### CODE REVIEW

**Comment by evverx** [comment]
I've just found the following code in the `emacs` repo
```c
  /* Call 'personality' twice, to detect buggy platforms like WSL
     where 'personality' always returns 0.  */
  return (pers != desired_pers
          && personality (desired_pers) == pers
          && personality (0xffffffff) == desired_pers);
```
I hope nobody runs `systemd` on `WSL` :-)

File: src/test/test-seccomp.c [hunk 1]
```diff
@@ -582,59 +582,23 @@ static void test_lock_personality(void) {
...
+                assert_se(safe_personality(PER_LINUX | READ_IMPLIES_EXEC) == -EPERM);
+                assert_se(safe_personality(PER_LINUX_32BIT) == -EPERM);
+                assert_se(safe_personality(PER_SVR4) == -EPERM);
+                assert_se(safe_personality(PER_BSD) == -EPERM);
+                assert_se(safe_personality(current == PER_LINUX ? PER_LINUX32 : PER_LINUX) == -EPERM);
+                assert_se(safe_personality(PER_LINUX32_3GB) == -EPERM);
+                assert_se(safe_personality(PER_UW7) == -EPERM);
+                assert_se(safe_personality(0x42) == -EPERM);
+
+                assert_se(safe_personality(PERSONALITY_INVALID) == -EPERM); /* maybe remove this later */
```
Comment (evverx) [inline src/test/test-seccomp.c:606]: I think it would make sense to check that `errno` is set when `personality` fails. Not that it matters.

**Comment by tixxdz** [comment]
lgtm, apart that comment from @evverx for WSL, I don't know this stuff and systemd has too much Linux specific stuff... however it seems they use WSL for containers ? and googled this issue:
https://github.com/Microsoft/BashOnWindows/issues/994

so maybe lets be nice and remove one issue from the road...

**Comment by poettering** [comment]
We really don't care about WSL, and if it's broken, then Microsoft needs to fix it, we won't work-around it.

Anyway, I pushed a new version now, adding the extra errno check @evverx suggested. Please have another look and merge!

**Comment by evverx** [comment]
@poettering, thank you. LGTM.

@tixxdz, thank you for the link to https://github.com/Microsoft/BashOnWindows/issues/994. I wasn't aware of the fact that people really try to run `systemd` on `WSL`. I think it will be an interesting thread to read :-)

**Comment by evverx** [comment]
The failure seems to be unrelated to this PR:
```
======================================================================
ERROR: test_hotplug_dhcp_ip6 (__main__.DnsmasqClientTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/adttmp/autopkgtest-virt-lxc.shared.k8ukdh2w/downtmp/build.OSZ/debian/test/networkd-test.py", line 405, in test_hotplug_dhcp_ip6
    self.do_test(coldplug=False, ipv6=True)
  File "/data/adttmp/autopkgtest-virt-lxc.shared.k8ukdh2w/downtmp/build.OSZ/debian/test/networkd-test.py", line 308, in do_test
    self.iface, '--timeout=%i' % online_timeout])
  File "/usr/lib/python3.5/subprocess.py", line 581, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/lib/systemd/systemd-networkd-wait-online', '--interface', 'test_eth42', '--timeout=10']' returned non-zero exit status 1

----------------------------------------------------------------------
Ran 35 tests in 265.778s
```

**Pull request merged**